ymer/src/main/java/com/avanza/ymer/PersistedInstanceIdRecalculationService.java

```diff
@@ -57,21 +60,27 @@ public void recalculatePersistedInstanceId(String collectionName, int numberOfIn
 		if (!indexDropped) {
 			log.info("Step 1/3\tNo index to drop");
 		}
-		AtomicInteger count = new AtomicInteger();
-		collection.findByQuery(query(where(DOCUMENT_ROUTING_KEY).exists(true)))
-				.peek(it -> {
-					int currentCount = count.incrementAndGet();
-					if (currentCount % 10_000 == 0) {
-						log.info("Step 2/3\tUpdated persisted instance id for {} documents", currentCount);
-					}
-				})
-				.forEach(document -> {
-					Object id = document.get("_id");
-					int instanceId = getInstanceId(document.get(DOCUMENT_ROUTING_KEY), numberOfInstances);
-					collection.updateById(id, Map.of(DOCUMENT_INSTANCE_ID, instanceId));
-				});
 
-		log.info("Step 2/3\tUpdated persisted instance id for {} documents", count.get());
+		try (Stream<List<Document>> batches = collection.findByQuery(query(where(DOCUMENT_ROUTING_KEY).exists(true)), BATCH_SIZE)) {
```

Comment:
Do you think it would be worthwhile to add a projection here so that only the affected fields are returned from the query instead of the whole documents?

Reply:
👍

Reply:
Good idea! 👍